plugins/httpapi/mso.py

```diff
@@ -236,33 +236,38 @@ def _response_to_json(self, response_data):
         # JSONDecodeError only available on Python 3.5+
         except Exception as e:
             # Expose RAW output for troubleshooting
-            self.error = dict(code=-1, message="Unable to parse output as JSON, see 'raw' output. {0}".format(e))
+            self.error = dict(code=self.status, message="Unable to parse output as JSON, see 'raw' output. {0}".format(e))
             self.info["raw"] = response_text
             return
 
     def _get_login_domain_id(self, domain_name):
         """Get a domain and return its id"""
         if domain_name is None:
             return None
+
+        method = "GET"
+        path = "/mso/api/v1/auth/login-domains"
+        full_path = self.connection.get_option("host") + path
+
         # TODO: Replace response by -
-        response, data = self.send_request("GET", "auth/login-domains")
+        response, data = self.connection.send(path, None, method=method, headers=self.headers)
 
         if data is not None:
```

Comment:
only during not None? is there option of [] return? what would then be behaviour? error?

Reply:
Irrespective of what is returned, the general idea is make an attempt to parse the data as JSON. The _response_to_json() function takes care of any exceptions that may be encountered.

Reply:
ok but if data is [] it is valid data and would do execution of _response_to_json() which is not needed since it is empty list? would if data be sufficient? an error should still be raised in that case I think.